### PR TITLE
TYPO3 10 LTS compatible

### PR DESCRIPTION
**Pull request by tomasnorre**

### CODE REVIEW

File: Classes/TYPO3/Hooks/BackendAjaxHook.php [hunk 1]
```diff
@@ -54,7 +50,17 @@ public function banAll(ServerRequestInterface $request, ResponseInterface $respo
...
+        // We need to return a response to satisfy the
+        // TYPO3\CMS\Backend\Http\RouteDispatcher::dispatch()
+        // don't like the solution, but haven't come up with something better yet.
+        return new Response();
+    }
+
+    protected function getVarnish(): Varnish
+    {
+        $objectManager = GeneralUtility::makeInstance(ObjectManager::class);
+        $this->varnish = $objectManager->get(Varnish::class);
```
Comment (tomasnorre) [inline Classes/TYPO3/Hooks/BackendAjaxHook.php:62]: I know that the ObjectManager->get() is deprecated, but haven't found a better way to do this in context of a Backend Hook. The GeneralUtility::makeInstance() isn't working in this context for the Varnish instance.

**Review by TomSchenk — APPROVED** [review]

File: Classes/TYPO3/Hooks/BackendAjaxHook.php (hunk 1 above)
Comment (tlayh) [reply Classes/TYPO3/Hooks/BackendAjaxHook.php:62]: AbstractHook is using the variant with GeneralUtility::makeInstance(). Is it different there?

File: Classes/System/Http.php [hunk 2]
```diff
@@ -29,6 +29,7 @@
 use GuzzleHttp\Promise\PromiseInterface;
 use GuzzleHttp\Psr7\Response;
 use GuzzleHttp\RequestOptions;
+use function GuzzleHttp\Promise\settle;
```
Comment (tlayh) [inline Classes/System/Http.php]: settle is deprecated. Is there some other way? Utils::settle ?

File: Tests/Unit/TYPO3/Hooks/ContentPostProcOutputHookTest.php [hunk 3]
```diff
@@ -29,6 +29,9 @@
 use Aoe\Varnish\TYPO3\Configuration\ExtensionConfiguration;
 use Aoe\Varnish\TYPO3\Hooks\ContentPostProcOutputHook;
 use Nimut\TestingFramework\TestCase\UnitTestCase;
+use TYPO3\CMS\Core\Cache\Backend\NullBackend;
+use TYPO3\CMS\Core\Cache\CacheManager;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
 use TYPO3\CMS\Extbase\Object\ObjectManagerInterface;
 use TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController;
```
Comment (tlayh) [inline Tests/Unit/TYPO3/Hooks/ContentPostProcOutputHookTest.php:33]: Some unused imports here.

File: Tests/Unit/TYPO3/Hooks/TceMainHookTest.php [hunk 4]
```diff
@@ -31,7 +31,10 @@
 use Aoe\Varnish\TYPO3\Hooks\TceMainHook;
 use Nimut\TestingFramework\TestCase\UnitTestCase;
 use TYPO3\CMS\Core\Authentication\BackendUserAuthentication;
+use TYPO3\CMS\Core\Cache\Backend\NullBackend;
+use TYPO3\CMS\Core\Cache\CacheManager;
 use TYPO3\CMS\Core\DataHandling\DataHandler;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
 use TYPO3\CMS\Extbase\Object\ObjectManagerInterface;
```
Comment (tlayh) [inline Tests/Unit/TYPO3/Hooks/TceMainHookTest.php:38]: ObjectManagerInterface not used anymore

**Review by tlayh — CHANGES_REQUESTED** [review]

File: Classes/System/Http.php (hunk 2 above)
Comment (tomasnorre) [reply Classes/System/Http.php]: Yep. Didn't pay attention to this.

**Pull request merged**